File: src/client/src/api/schedules/scheduleResponse.ts
import type { AudienceResponse } from '@/api/audiences/audienceResponse'
import type { TimeslotResponse } from '@/api/schedules/timeslots/timeslotResponse'

export interface ScheduleResponse {
  id: string
  startsAt: string
  endsAt: string
  description: string
  audience: AudienceResponse
  timeslots: TimeslotResponse[]
  isEditable: boolean
  isDeletable: boolean
  isTimeslotCreationAllowed: boolean
}

File: src/server/LowPressureZone.Api/Endpoints/Schedules/ScheduleResponse.cs
using LowPressureZone.Api.Endpoints.Communities;
using LowPressureZone.Core;

namespace LowPressureZone.Api.Endpoints.Schedules;

public sealed class ScheduleResponse : ITimeRange
{
    public required Guid Id { get; set; }
    public required DateTimeOffset StartsAt { get; set; }
    public required DateTimeOffset EndsAt { get; set; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public required CommunityResponse Community { get; set; }
    public required IEnumerable<ITimeRange> Slots { get; set; }
    public required bool IsEditable { get; set; }
    public required bool IsDeletable { get; set; }
    public required bool IsHourlyAllowed { get; set; }
    public required bool IsClashAllowed { get; set; }
    public required bool IsVisibleToPublic { get; set; }
}